src/s2geography/geoarrow.cc

```diff
@@ -792,6 +804,24 @@ class WriterImpl {
     return GEOARROW_OK;
   }
 
+  int VisitPolylineEdges(const S2Polyline& poly) {
+
+    for (int i = 1; i < poly.num_vertices(); i++) {
+        const S2Point& pt0(poly.vertex(i - 1));
+        const S2Point& pt1(poly.vertex(i));
+        tessellator_->AppendProjected(pt0, pt1, &points_);
```

Comment:
Do we always want to go through the tessellator `AppendProjected`? 
Or in case of no tessellation (set to infinity), better to directly use the projection's `Project` just per vertex?

(in any case, for VisitPoint I still have to add the Project)